acvm-repo/acvm/src/pwg/brillig.rs

```diff
@@ -200,16 +200,16 @@ impl<'b, B: BlackBoxFunctionSolver<F>, F: AcirField> BrilligSolver<'b, F, B> {
     }
 
     pub(crate) fn finalize(
```

Comment:
💭  There could be two finalize methods:
- `finalize` (the existing one before this PR)
- `finalize_with_profiling` that's the new one in this PR

That way we'd avoid the many `.finalize().0` calls. Similarly, there could be `execute_program` and `execute_program_with_profiling` to avoid some `.map(|(witness_stack, _)| witness_stack)`.

Reply:
done